static/js/click_handlers.js

```diff
@@ -991,8 +991,13 @@ export function initialize() {
         }
 
         if (compose_state.composing()) {
-            if ($(e.target).closest("a").length > 0) {
-                // Refocus compose message text box if link is clicked
+            if (
+                $(e.target).closest("a").length > 0 &&
+                !$(e.target).closest("a").hasClass("new_topic_stream")
+            ) {
+                // Refocus compose message text box if one clicks an external
+                // link/url to view something else while composing a message
+                // See issue #4331 for more details
```

Comment:
My guess is that the actual fix is to do an `e.preventDefault()` in the click handler rather than changing this code.

Reply:
I have removed the change I did. The `e.preventDefault()` wasn't needed the `e.stopPropagation` works. Actually this code was a lot with the focus of the text areas. Actually if you click any link, then this thinks that it needs to focus the textarea. So, I worked on days to find this and I talked to @showell he suggested improving the comment. So, I have not changed the code, but changed the comment as it is very necessary.
Also he suggested, if you want to you can make the comment change in a different commit.

Reply:
Generally it's good to use both preventDefault and stopPropagation whenever using one or the other, unless one has a good reason to do so.

Reply:
Do you think the comment change here is useful?

Reply:
Yes, according to me, the comment change is very useful. The previous comment didn't explain why this condition was there and was a bit hard to understand the real use case of it. This was a very useful feature which literally no one notices easily.

Reply:
Thanks. I am reading more about stopPropagation. Will do this one as soon as I can.

Reply:
@timabbott I think `e.preventDefault()` isn't needed as it is used to prevent the default behaviour. There isn't any default behaviour of this listener which we attached. In the case of `e.stopPropagation`, it stops the further propagation which was much needed which was manually fixed in the changed condition above.
Rest, it's just my views. There is no problem in using both, just preventing extra lines which may be not needed.

Reply:
It's better to just use them both for every handler, unless we specifically want the default browser behavior -- because not doing so has lead to a number of weird bugs in the past.  So I added `preventDefault` before testing and merging.

Reply:
Sure, I will! Thanks.